src/Utilities/ContainerHelpers.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Requires.hpp"
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Returns the `i`th element if `T` has a subscript operator, otherwise
+ * if `T` is fundamental returns `t`.
+ */
+template <class T>
+SPECTRE_ALWAYS_INLINE auto get_element(const T& t, const size_t i)
```

Comment:
noexcept everything

Reply:
These are forced to be inline, they can't throw exceptions

Reply:
In the sense that the function that calls them handles it

Reply:
For the record, this does affect the generated code on clang.  It inlines the terminate check along with the function body.  This does not appear to be true of gcc.